manager-bundle/src/Composer/ScriptHandler.php

```diff
@@ -18,107 +18,70 @@
 use Symfony\Component\Process\Process;
 use Webmozart\PathUtil\Path;
 
+/**
+ * @deprecated Deprecated since Contao 4.11, to be removed in Contao 5.0; use
+ *             the vendor/bin/contao-setup binary instead.
+ */
 class ScriptHandler
 {
     /**
      * Runs all Composer tasks to initialize a Contao Managed Edition.
      */
     public static function initializeApplication(Event $event): void
     {
-        $webDir = self::getWebDir($event);
-
-        static::purgeCacheFolder();
-        static::executeCommand(['contao:install-web-dir'], $event);
-        static::executeCommand(['cache:clear', '--no-warmup'], $event);
-        static::executeCommand(['cache:clear', '--no-warmup'], $event, 'dev');
-        static::executeCommand(['cache:warmup'], $event);
-        static::executeCommand(['assets:install', $webDir, '--symlink', '--relative'], $event);
-        static::executeCommand(['contao:install', $webDir], $event);
-        static::executeCommand(['contao:symlinks', $webDir], $event);
-
-        $event->getIO()->write('<info>Done! Please open the Contao install tool or run contao:migrate on the command line to make sure the database is up-to-date.</info>');
-    }
-
-    public static function purgeCacheFolder(): void
-    {
-        $filesystem = new Filesystem();
-        $filesystem->removeDirectory(Path::join(getcwd(), 'var/cache/prod'));
-    }
-
-    /**
-     * Adds the app directory if it does not exist.
-     */
-    public static function addAppDirectory(): void
-    {
-        $filesystem = new Filesystem();
-        $filesystem->ensureDirectoryExists(Path::join(getcwd(), 'app'));
-    }
-
-    /**
-     * @throws \RuntimeException
-     */
-    private static function executeCommand(array $cmd, Event $event, string $env = 'prod'): void
-    {
-        $phpFinder = new PhpExecutableFinder();
-
-        if (false === ($phpPath = $phpFinder->find())) {
-            throw new \RuntimeException('The php executable could not be found.');
-        }
-
-        $command = array_merge(
-            [$phpPath, __DIR__.'/../../bin/contao-console'],
-            $cmd,
-            ['--env='.$env, $event->getIO()->isDecorated() ? '--ansi' : '--no-ansi']
+        trigger_deprecation('contao/manager-bundle', '4.11', 'Using ScriptHandler::initializeApplication() has been deprecated and will no longer work in Contao 5.0. Use the vendor/bin/contao-setup binary instead.');
+
+        $command = array_filter([
+            'contao-setup',
+            $event->getIO()->isDecorated() ? '--ansi' : '--no-ansi',
+        ]);
+
+        $event->getIO()->write(
+            sprintf(
+                '<warning>Please edit your root composer.json and set "%s" to "%s" instead of using "ScriptHandler::initializeApplication()".</warning>',
+                $event->getName(),
+                implode(' ', $command)
```

Comment:
Will this output `vendor/bin/contao-setup` or just `contao-setup`?

Reply:
It will actually output `contao-setup --ansi` (or `--no-ansi`). 

If the argument is omitted - which should imo be the default in the managed edition's `composer.json` - the right mode is guessed.